File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def isNotPrime(num):
    if num < 2:
        return True
    
    for i in range(2, num):
            if num % i == 0:
                return True
    
    return False

def solution(numbers):
    _set = set()
    for i in range(1, len(numbers)+1):
        for cou in permutations(numbers, i):
            _set.add(int(''.join(cou)))
    
    ans = len(_set)
    for num in _set:
        if isNotPrime(num):
            ans -= 1
            continue
            
    return ans

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    
    int answer = 0;
    int size = 0;
    String[] num_arr = null;
    static Set<Integer> primeSet = new HashSet<>();
    
    public int solution(String numbers) {
        this.num_arr = numbers.split("");
        this.size = num_arr.length;
        
        dfs(0, "", new boolean[size]);
        
        return this.answer;
    }
    
    public void dfs(int n, String num, boolean[] visited){
        if(n == size){
            if(!num.isEmpty()){
                int target = Integer.parseInt(num);
                if(!primeSet.contains(target) && isPrime(target)){
                    this.answer++;
                    primeSet.add(target);
                }
            }
            return;
        }
        
        for(int i = 0; i < this.size; i++){
            if(visited[i] == false){
                visited[i] = true;
                dfs(n+1, num + num_arr[i], visited);
                visited[i] = false;
                dfs(n+1, num, visited);
            }
        }
    }
    
    public boolean isPrime(int num){
        if (num < 2) return false;
        for(int i = 2; i <= Math.sqrt(num); i++){
            if (num % i == 0) return false;
        }
        return true;
    }
    
}